mapper/tedge_mapper/tests/c8y_threshold_packet_size.rs

```diff
@@ -0,0 +1,150 @@
+use futures::future::TryFutureExt;
+use librumqttd::{async_locallink, Config};
+use mqtt_client::{Client, Message, MqttClient, QoS, Topic, TopicFilter};
+use std::process::Command;
+use std::{thread, time};
+
+#[derive(Debug)]
+enum TestJoinError {
+    TestThresholdError,
+    ElapseTime,
+}
+
+#[tokio::test]
+// This checks the mqtt packet size that exceeds the limit
+async fn c8y_threshold_packet_size() -> Result<(), anyhow::Error> {
+    // Start the broker
+    let mqtt_server_handle = tokio::spawn(async {
+        start_broker_local("../../configuration/rumqttd/rumqttd_5885.conf").await
+    });
+
+    // set the port
+    setup();
+
+    // wait for c8y mapper to start
+    let ten_millis = time::Duration::from_secs(1);
+    thread::sleep(ten_millis);
+
+    // Start the publisher and publish a message
+    let publish = tokio::spawn(async { publish_big_message_wait_for_error().await });
+
+    // if error is received then test is ok, else test should fail
+    let res = publish.await?;
+    mqtt_server_handle.abort();
+    match res {
+        Err(e) => {
+            cleanup();
+            return Err(e);
+        }
+        _ => {
+            cleanup();
+            return Ok(());
+        }
+    }
+}
+
+async fn subscribe_errors(pub_client: &Client) -> Result<(), anyhow::Error> {
+    let error_filter = TopicFilter::new("tedge/errors")?;
+    let mut error_messages = pub_client.subscribe(error_filter).await?;
+    while let Some(error) = error_messages.next().await {
+        let payload_str: String = error.payload_str().unwrap().into();
+        if payload_str.contains("The input size 20480 is too big. The threshold is 16384") {
+            anyhow::bail!("The input size 20480 is too big. The threshold is 16384");
+        } else {
+            return Ok(());
+        }
+    }
+    Ok(())
+}
+
+async fn start_broker_local(cfile: &str) -> anyhow::Result<()> {
+    let config: Config = confy::load_path(cfile)?;
+    let (mut router, _console, servers, _builder) = async_locallink::construct_broker(config);
+    let router = tokio::task::spawn_blocking(move || -> anyhow::Result<()> { Ok(router.start()?) });
+    servers.await;
+    let _ = router.await;
+    Ok(())
+}
+
+async fn publish_big_message_wait_for_error() -> Result<(), anyhow::Error> {
+    // create a 20KB message
+    let buffer = create_packet(1024 * 20);
```

Comment:
I believe I already saw this functions somewhere...
Maybe you could move it to `tedge_utils` crate?

Reply:
Sure, this is already under another refactoring PR. I will do it as part of that.